api/valor_api/backend/metrics/classification.py

```diff
@@ -529,48 +545,31 @@ def _compute_roc_auc(
 
     sum_roc_aucs = 0
     label_count = 0
+    for _, label_rows in value_to_labels_mapping.items():
+        for label_row in label_rows:
 
-    for grouper_value, labels in value_to_labels_mapping.items():
-        label_filter = groundtruth_filter.model_copy()
-        label_filter.labels = schemas.LogicalFunction.and_(
-            label_filter.labels,
-            schemas.LogicalFunction.or_(
-                *[
-                    schemas.Condition(
-                        lhs=schemas.Symbol(
-                            name=schemas.SupportedSymbol.LABEL_ID
-                        ),
-                        rhs=schemas.Value.infer(label.id),
-                        op=schemas.FilterOperator.EQ,
-                    )
-                    for label in labels
-                ]
-            ),
-        )
-
-        # some labels in the "labels" argument may be out-of-scope given our groundtruth_filter, so we fetch all labels that are within scope of the groundtruth_filter to make sure we don't calculate ROCAUC for inappropriate labels
-        in_scope_labels = [
-            label
-            for label in labels
-            if schemas.Label(key=label.key, value=label.value)
-            in core.get_labels(
-                db=db, filters=label_filter, ignore_predictions=True
-            )
-        ]
+            label = schemas.Label(key=label_row.key, value=label_row.value)
 
-        if not in_scope_labels:
-            continue
+            # some labels in the "labels" argument may be out-of-scope given our groundtruth_filter, so we fetch all labels that are within scope of the groundtruth_filter to make sure we don't calculate ROCAUC for inappropriate labels
+            if label not in filtered_label_set:
+                continue
+            elif (
+                label not in label_to_count
+                or label.key not in label_key_to_count
+            ):
+                raise RuntimeError("ROCAUC computation failed.")
```

Comment:
nit: could this error message be made more specific?

Reply:
Updated to `ROC AUC computation failed as the label `{label}` could not be found.`